apps/studio/src/features/editing-experience/hooks/useLinkEditorModal.tsx

```diff
@@ -0,0 +1,47 @@
+import type { Editor } from "@tiptap/react"
+import { useDisclosure } from "@chakra-ui/react"
+
+import { LinkEditorModal } from "~/components/PageEditor/LinkEditorModal"
+
+interface UseLinkEditorModalProps {
+  editor: Editor
+}
+
+export const useLinkEditorModal = ({ editor }: UseLinkEditorModalProps) => {
+  const {
+    isOpen: isLinkModalOpen,
+    onOpen: onLinkModalOpen,
+    onClose: onLinkModalClose,
+  } = useDisclosure()
+
+  return {
+    onLinkModalOpen,
+    LinkEditorModal: (
+      <LinkEditorModal
+        linkText={
+          editor.isActive("link")
+            ? editor.state.doc.nodeAt(
+                Math.max(1, editor.view.state.selection.from - 1),
+              )?.textContent
+            : ""
```

Comment:
i haven't actually taken a look at this, but why would `from` not be sufficient ah? also, if no selection (single I cursor, haven't select anything yet), how would this work?

Reply:
The `from` takes the position just after the actual `from` (implicit +1), so subtracting 1 here to actually get the cursor's position.